x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/phases/shared/data_tier_allocation_field.tsx

```diff
@@ -48,9 +49,32 @@ export const DataTierAllocationField: FunctionComponent<Props> = ({
   return (
     <NodesDataProvider>
       {(nodesData) => {
-        const isCompatible = isPhaseDefaultDataAllocationCompatible(phase, nodesData.nodesByRoles);
         const hasNodeAttrs = Boolean(Object.keys(nodesData.nodesByAttributes ?? {}).length);
 
+        const renderDefaultAllocationNotice = () => {
```

Comment:
WDYT of organizing the early exits first, to help the reader scan and understand the states that don't require a notice? Something like this maybe:

```js
const renderDefaultAllocationNotice = () => {
  if (phaseData.dataTierAllocationType !== 'default') {
    return null;
  }

  const allocationNodeRole = determineAllocationNodeRole(phase, nodesData.nodesByRoles);
  const isFirstPreferredNodeRole = allocationNodeRole !== 'none' && isNodeRoleFirstPreference(phase, allocationNodeRole);
  
  if (isFirstPreferredNodeRole) {
    return null;
  }

  return <DefaultAllocationNotice phase={phase} targetNodeRole={allocationNodeRole} />;
};
```